javascript/frameworks/cap/lib/advanced_security/javascript/frameworks/cap/RemoteFlowSources.qll

```diff
@@ -64,3 +68,46 @@ class ServiceinCDSHandlerParameter extends ParameterNode, RemoteFlowSource {
     result = "Parameter of an event handler belonging to an exposed service defined in a cds file"
   }
 }
+
+/**
+ * A parameter of a handler registered for a service on an event. e.g.
+ * ```javascript
+ * cds.serve('./test-service').with((srv) => {
+ *  srv.before('READ', '*', (req) => req.reply([]))
+ * })
+ * ```
+ * The parameter named `req` is captured in the above example.
+ *
+ * DOES NOT REQUIRE that a `UserDefinedApplicationService` is explicitly defined
+ * DOES NOT REQUIRE that the name is provided explicitly
+ */
+class HandlerParameterImplicitService extends ParameterNode, RemoteFlowSource {
+  Handler handler;
+  HandlerRegistration handlerRegistration;
+
+  HandlerParameterImplicitService() {
+    exists(ServiceInstanceFromServeWithParameter service |
+      handler = handlerRegistration.getHandler() and
+      this = handler.getParameter(0) and
+      service.getAHandlerRegistration() = handlerRegistration and
+      //this will otherwise duplicate on the case where we do actually know the
+      //name from the cds file and it matches up
+      //only relevant if you are using the specific type anyhow (as opposed to RemoteFlowSource)
+      not this instanceof ServiceinCDSHandlerParameterWithName
```

Comment:
It might be helpful to add an example snippet in the comment. Also, block comments are preferred when providing multiline comments.